src/main/java/software/amazon/event/kafkaconnector/batch/BestEffortInOrderEventBridgeBatching.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.batch;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Stream;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+
+public class BestEffortInOrderEventBridgeBatching implements EventBridgeBatchingStrategy {
+
+  private final Collector<
+          MappedSinkRecord<PutEventsRequestEntry>,
+          PutEventsRequestEntrySizeCollector,
+          Stream<List<MappedSinkRecord<PutEventsRequestEntry>>>>
+      collector =
+          Collector.of(
+              PutEventsRequestEntrySizeCollector::new,
+              PutEventsRequestEntrySizeCollector::accumulate,
+              (left, right) -> {
+                throw new IllegalStateException();
+              },
+              PutEventsRequestEntrySizeCollector::finish);
+
+  @Override
+  public Stream<List<MappedSinkRecord<PutEventsRequestEntry>>> apply(
+      Stream<MappedSinkRecord<PutEventsRequestEntry>> records) {
+    if (records.isParallel()) {
+      throw new IllegalArgumentException("Stream must not be parallel.");
+    }
+    return records.collect(collector);
+  }
+
+  static class PutEventsRequestEntrySizeCollector {
+
+    private static final int MAX_BATCH_SIZE = 256 * 1024;
+
+    private final List<List<MappedSinkRecord<PutEventsRequestEntry>>> result = new ArrayList<>();
+    private int actualBatchSize = 0;
+    private int index = 0;
+
+    PutEventsRequestEntrySizeCollector() {
+      result.add(new ArrayList<>());
```

Comment:
Q: any concerns with initializing result here again with new ArrayList since it's also done in line 44?

Reply:
In line 44 the result list is initialized. In line 49 the first (empty) batch is added to not check the state in the `accumulate` method call. So I can access the result list even at index 0 without testing the size.